File: fragment/fragment/src/main/java/com/laioffer/demo1/Demo1Activity.java
package com.laioffer.demo1;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.FragmentActivity;

import com.laioffer.fragment.fragment.R;

/**
 * 类的描述: demo1:Fragment基本使用
 * Created by 春夏秋冬在中南 on 2023/6/8 23:21
 */
public class Demo1Activity extends FragmentActivity {

  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_demo1);
  }
}

File: jetpack/src/main/java/com/laioffer/lesson2_livedata/demo1/Demo1Activity.kt
package com.laioffer.lesson2_livedata.demo1

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.laioffer.R
import kotlin.concurrent.thread

/**
 * 类的描述: LiveData简单使用
 *
 * Created by 春夏秋冬在中南 on 2023/9/16 21:10
 */
class Demo1Activity : AppCompatActivity() {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.single_tv_layout)

    val textView: TextView = findViewById(R.id.tv_textview)

    // 1 观察者 眼睛 环节
    MyLiveData.info1.observe(this) {
      textView.text = it // 更新UI
    }

    // 完整写法 new  Observer  onChanged
    // MyLiveData.info1.observe(this, object : Observer<String> {
    //   override fun onChanged(t: String?) {
    //     textView.text = t // 更新UI
    //   }
    // })


    // 2 触发数据改变 环节
    MyLiveData.info1.value = "default" // setValue 主线程

    thread {
      Thread.sleep(3000)
      MyLiveData.info1.postValue("3秒钟后，修改了哦")  // postValue 子线程
    }

    thread { // 子线程
      Thread.sleep(6000)
      MyLiveData.info1.postValue("6秒钟后，修改了哦")  // postValue 子线程
    }
  }

}